sandbox/cgroup/tracking.go

```diff
@@ -487,3 +491,72 @@ var doCreateTransientScope = func(conn *dbus.Conn, unitName string, pid int) err
 // of avoiding clashing UUIDs that are needed for all of the non-service
 // commands that are started with the help of this UUID.
 var randomUUID = randutil.RandomKernelUUID
+
+// SnapDeviceCgroupOptions represents the options for device cgroup tracking.
+type SnapDeviceCgroupOptions struct {
+	NonStrict   bool // NonStrict being true implies the application is installed with --devmode
+	SelfManaged bool // SelfManaged ensures that no device management eBPF program is loaded for this snap.
+}
+
+// LoadSnapDeviceCgroupOptions loads the device cgroup options for a given security tag.
+func LoadSnapDeviceCgroupOptions(securityTag string) (opts SnapDeviceCgroupOptions, err error) {
+	path := filepath.Join(dirs.SnapCgroupPolicyDir, fmt.Sprintf("%s.device", securityTag))
```

Comment:
This is duplicating the code for the path in `snapDeviceCgroupSelfManageFilePath()`, this probably needs to be an ancillary method somewhere to ensure one root of trust for building the path.